src/lib/storage/chunk.cpp

```diff
@@ -140,6 +140,12 @@ std::shared_ptr<BaseIndex> Chunk::get_index(const ColumnIndexType index_type,
   return get_index(index_type, columns);
 }
 
+void Chunk::remove_index(std::shared_ptr<BaseIndex> index) {
+  auto it = std::find(_indices.cbegin(), _indices.cend(), index);
+  if (it == _indices.cend()) Fail("Trying to remove a non-existing index");
```

Comment:
Consider `(Debug)Assert()`

Reply:
+1 for `DebugAssert`. This would be more consistent with other methods like this (`Table::column_is_nullable`, `Table::column_type`, ...)

```c++
bool Table::column_is_nullable(ColumnID column_id) const {
  DebugAssert(column_id < _column_names.size(), "ColumnID " + std::to_string(column_id) + " out of range");
  return _column_nullable[column_id];
}
```

Reply:
good point!